fabric-network/src/impl/query/roundrobinqueryhandler.js

```diff
@@ -13,52 +13,46 @@ const util = require('util');
 const logger = require('fabric-network/lib/logger').getLogger('RoundRobinQueryHandler');
 
 class RoundRobinQueryHandler {
-	constructor(peers, options = {}) {
+	constructor(peers) {
 		logger.debug('constructor: peers=%j', peers.map((peer) => peer.name));
 		this._peers = peers;
 		this._currentPeerIndex = 0;
-		this._options = options;
 	}
 
 	async evaluate(query) {
 		const method = 'evaluate';
 		logger.debug('%s - start', method);
 
-		const startPeerIndex = this._currentPeerIndex;
-		const errorMessages = [];
-		const options = {requestTimeout: 3000};
-		// use the timeout as the requestTimeout or let default
-		if (Number.isInteger(this._options.timeout)) {
-			options.requestTimeout = this._options.timeout * 1000; // in ms;
+		if (this._currentPeerIndex >= this._peers.length) {
```

Comment:
This is a good catch. The previous code could eventually have reached MAX_SAFE_INTEGER and misbehaved. I wonder if it's clearer as something like:
```
const startPeerIndex = this._currentPeerIndex;
this._currentPeerIndex = (this._currentPeerIndex + 1) % this._peers.length;
```

Reply:
yes, I like it